tests/suites/test_suite_ssl.function

```diff
@@ -5102,9 +5102,7 @@ void inject_client_content_on_the_wire(int pk_alg,
         ret = mbedtls_ssl_handshake_step(&server.ssl);
     } while (ret == 0 && server.ssl.state == state);
     TEST_EQUAL(ret,  expected_ret);
-    /* If we're expected to suceeed and we do, that's enough.
-     * If we're expected to fail, also check it was in the expected way. */
-    if (expected_ret != 0) {
+    if (strlen(log_pattern) != 0) {
         TEST_EQUAL(srv_pattern.counter, 1);
     }
```

Comment:
Minor: why does it matter that the pattern was seen only once? Why not
```suggestion
    TEST_LE_U(1, srv_pattern.counter);
```

Reply:
It does not really matter - it just so happened that until now the patterns I was testing were only appearing once, so I went with the strictest assert, but indeed it we start checking logs in all cases we'll probably want to allow more than one match.